src/node.cc

```diff
@@ -3901,9 +3906,7 @@ Environment* CreateEnvironment(Isolate* isolate,
       HandleCleanup,
       nullptr);
 
-  if (v8_is_profiling) {
-    StartProfilerIdleNotifier(env);
-  }
+  StartProfilerIdleNotifier(env);
```

Comment:
It registers SetIdle and ClearIdle with libuv, and those only set the vm state enum value. Looks rather very small (trivial) overhead.